web-api/src/business/useCases/courtIssuedDocument/fileAndServeCourtIssuedDocumentInteractor.deadlines.test.ts

```diff
@@ -0,0 +1,255 @@
+  let mockWorkItem;
+  let mockDocketEntryWithWorkItem;
+  let mockDeadline;
+  let consolidatedCaseExample;
+
+  const mockPdfUrl = 'www.example.com';
+  const mockClientConnectionId = 'ABC123';
+  const mockDocketEntryId = 'c54ba5a9-b37b-479d-9201-067ec6e335ba';
+
+  beforeEach(() => {
+    MOCK_DATE = '2022-12-06T22:54:06.000Z';
+
+    mockWorkItem = {
+      docketNumber: MOCK_CASE.docketNumber,
+      section: DOCKET_SECTION,
+      sentBy: docketClerkUser.name,
+      sentByUserId: docketClerkUser.userId,
+      workItemId: 'b4c7337f-9ca0-45d9-9396-75e003f81e32',
+    };
+
+    mockDocketEntryWithWorkItem = {
+      docketEntryId: mockDocketEntryId,
+      docketNumber: MOCK_CASE.docketNumber,
+      documentTitle: 'Order',
+      documentType: 'Order',
+      eventCode: 'O',
+      signedAt: '2019-03-01T21:40:46.415Z',
+      signedByUserId: docketClerkUser.userId,
+      signedJudgeName: 'Dredd',
+      userId: docketClerkUser.userId,
+      workItem: mockWorkItem,
+    };
+
+    caseRecord = {
+      ...MOCK_CASE,
+      associatedJudge: judgeUser.name,
+      docketEntries: [
+        mockDocketEntryWithWorkItem,
+        {
+          docketEntryId: 'c54ba5a9-b37b-479d-9201-067ec6e335bc',
+          docketNumber: MOCK_CASE.docketNumber,
+          documentTitle: 'Order to Show Cause',
+          documentType: 'Order to Show Cause',
+          eventCode: 'OSC',
+          signedAt: '2019-03-01T21:40:46.415Z',
+          signedByUserId: docketClerkUser.userId,
+          signedJudgeName: 'Dredd',
+          userId: docketClerkUser.userId,
+        },
+        {
+          docketEntryId: '7f61161c-ede8-43ba-8fab-69e15d057012',
+          docketNumber: MOCK_CASE.docketNumber,
+          documentTitle: 'Transcript of [anything] on [date]',
+          documentType: 'Transcript',
+          eventCode: TRANSCRIPT_EVENT_CODE,
+          userId: docketClerkUser.userId,
+        },
+      ],
+    };
+
+    mockDeadline = {
+      caseDeadlineId: 'test-deadline-id',
+      deadlineDate: '2030-01-20T00:00:00.000Z',
+      description: 'Test Deadline',
+    };
+
+    consolidatedCaseExample = {
+      petitioners: [] as TPetitioner[],
+      caseCaption: '',
+      caseType: 'Deficiency' as CaseType,
+      createdAt: '',
+      partyType: '',
+      petitionPaymentStatus: '',
+      procedureType: '',
+      receivedAt: '',
+      status: '' as CaseStatus,
+      sortableDocketNumber: 0,
+      docketEntries: [],
+      hearings: [],
+      correspondence: [],
+      consolidatedCases: [],
+      docketNumberWithSuffix: '',
+    };
+
+    getUserById.mockResolvedValue(docketClerkUser as DbUser);
+
+    getCaseByDocketNumber.mockResolvedValue(caseRecord);
+    getCasesByDocketNumbers.mockResolvedValue([caseRecord]);
+
+    fileAndServeDocumentOnOneCase.mockImplementation(
+      ({ caseEntity }) => caseEntity,
+    );
+
+    applicationContext
+      .getUseCaseHelpers()
+      .serveDocumentAndGetPaperServicePdf.mockReturnValue({
```

Comment:
Can we do `mockResolvedValue` for async functions? It might seem unimportant, but sometimes this can lead to a test not actually testing what you think (e.g., a Promise is not falsey).